app/assets/stylesheets/layout/_top_menu.sass

```diff
@@ -166,103 +166,117 @@ $search-input-width-expanded: 300px
     left: 260px
     @include varprop(color, font-color-on-primary)
 
-.top-menu-search
-  display: inline-block
-  position: relative
-  height: $header-height
-  line-height: $header-height
-  margin: 0 15px
-  > div
-    padding-top: 9px
 
 .top-menu-search--wrapper
   float: left
 
-.top-menu-search--button
-  line-height: $header-height
+.top-menu-search
+  display: flex
+  align-items: center
+  position: relative
   height: $header-height
-  padding: 0 5px
-  @include varprop(font-size, header-item-font-size)
-  font-weight: normal
-  background: transparent
-  &:hover
-    text-decoration: none
+  line-height: $header-height
+  margin: 0 15px
 
-  .top-menu-search &
+  .top-menu-search--button
     position: absolute
-    top: 1px
-    right: 0
-    z-index: 10
+    right: 2px
+    @include varprop(font-size, header-item-font-size)
     @include varprop(color, header-item-font-color)
+    &:hover
+      text-decoration: none
 
-.top-menu-search--button.-input-focused
-  color: $header-search-field-font-color
-
-input[type="text"].top-menu-search--input
-  margin: 0
-  padding: 0 20px 0 10px
-  display: inline-block
-  border-width: 1px
-  border-style: solid
-  @include varprop(border-color, header-item-font-color)
-  border-radius: 4px
-  height: 31px
-  width: $search-input-width
-  outline: 0
-  font-size: 0.9rem
-  background: transparent
-  @include varprop(color, header-item-font-color)
-  -webkit-transition: width 0.2s ease-in-out
-  transition: width 0.2s ease-in-out
-  &::-ms-clear
-    margin-right: 5px
-    width: 20px
-  &:focus
-    background: white
-    border-width: 1px
-    border-style: solid
-    @include varprop(border-color, header-item-font-color)
+  .top-menu-search--button.-input-focused
     color: $header-search-field-font-color
-    &:hover
-      background: white
-  &:hover
-    background: unset
-    border-width: 1px
-    border-style: solid
-    @include varprop(border-color, header-item-font-color)
 
-input.top-menu-search--input.-expanded
-  width: $search-input-width-expanded
-  background: white
-  border-width: 1px
-  border-style: solid
-  @include varprop(border-color, header-item-font-color)
-  color: $header-search-field-font-color
-  &:hover
-    background: white
+  .top-menu-search--loading
+    top: $header-height - 11px  // display directly under ng-input field
+    height: 46px                // ng-option height + 1px border
+    z-index: 1051
+
+  .ng-select#q
```

Comment:
This selector looks weird to me as it consists of a class name and an ID. I would go for either or. And if you chose to use an ID please make it more explanative such as "#global-search-input".